src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -303,17 +303,19 @@ class LedgerImpl(vm: VM, blockchain: BlockchainImpl, blockchainConfig: Blockchai
     worldStateProxy.saveAccount(senderAddress, account.increaseBalance(-calculateUpfrontGas(stx.tx)).increaseNonce)
   }
 
-  private[ledger] def prepareProgramContext(stx: SignedTransaction, blockHeader: BlockHeader, worldStateProxy: InMemoryWorldStateProxy, config: EvmConfig): PC =
+  private[ledger] def prepareProgramContext(stx: SignedTransaction, blockHeader: BlockHeader, worldStateProxy: InMemoryWorldStateProxy,
+                                            config: EvmConfig): PC = {
     stx.tx.receivingAddress match {
       case None =>
         val address = worldStateProxy.createAddress(creatorAddr = stx.senderAddress)
-        val world1 = worldStateProxy.transfer(stx.senderAddress, address, UInt256(stx.tx.value))
-        ProgramContext(stx, address,  Program(stx.tx.payload), blockHeader, world1, config)
-
+        val worldAfterTransfer = worldStateProxy.transfer(stx.senderAddress, address, UInt256(stx.tx.value))
+        val savedClearedAccountWorld = worldAfterTransfer.resetAccount(address)
```

Comment:
This looks better! But I would go with a single method `initialiseAccount(address: Address, value: UInt256)` - that finds the balance of the possibly existing account (or zero), and adds `value` to it. Otherwise, just like `resetAccount`. The read/write vs 2 writes, is not that big of an issue since these should be cached. But the important thing is we have a single method that captures the logic, instead of duplicating it here and in `CREATE`

Reply:
Also please add a comment/scaladoc that it handles address conflicts.